addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3903,16 +3903,17 @@ export class OdooEditor extends EventTarget {
         this._pluginCall('cleanForSave', [element]);
 
         // Clean the zero-width spaces added by the `fillEmpty` function
-        // (flagged with the "data-oe-zws-empty-inline" attributes).
-        // Reverse the list to start from the deepest elements (for emptiness checks).
-        for (const emptyElement of element.querySelectorAll('[data-oe-zws-empty-inline]')) {
-            if (isZWS(emptyElement)) {
+        // (flagged with the "data-oe-zws-empty-inline" attributes). Reverse the
+        // list to start from the deepest elements (for emptiness checks).
+        for (const emptyElement of [...element.querySelectorAll('[data-oe-zws-empty-inline]')].reverse()) {
```

Comment:
The original comment was:
```js
// Clean the remaining ZeroWidthspaces added by the `fillEmpty` function
// ( contain "data-oe-zws-empty-inline" attr)
// If the element contain more than just a ZWS,
// we remove it and clean the attribute.
// If the element have a class,
// we only remove the attribute to ensure we don't break some style.
// Otherwise we remove the entire inline element.
```
I guess I changed it when I made the change, then reverted my change when rebasing but kept the comment as it was...